### PR TITLE
feat!: Add network name attribute to protocols

### LINKED ISSUE #227: [Admin] Add the network name as an attribute of a protocol
The dex network name must match the one from currencies.json, i.e. "OSMOSIS" and "NEUTRON". The network is necessary to uniquely identify currencies that pertain to the protocol.

### PR DESCRIPTION
**Pull request by KirilMihaylov**
Closes #227

### CODE REVIEW

File: platform/contracts/admin/src/contracts/protocol/mod.rs [hunk 1]
```diff
@@ -13,3 +16,10 @@ pub struct Protocol<T> {
     pub oracle: T,
     pub profit: T,
 }
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct ProtocolWithNetworkName {
```
Comment (gmanev7) [inline platform/contracts/admin/src/contracts/protocol/mod.rs:22]: this type should be named `Protocol` whereas the former one - `ProtocolTemplate`

File: platform/contracts/admin/src/contracts/protocol/mod.rs [hunk 2]
```diff
@@ -13,3 +16,10 @@ pub struct Protocol<T> {
...
+    pub network: String,
+    pub protocol: Protocol<Addr>,
```
Comment (gmanev7) [inline platform/contracts/admin/src/contracts/protocol/mod.rs:24]: `protocol` -> `contracts`

File: platform/contracts/admin/src/state/contracts.rs [hunk 3]
```diff
@@ -36,15 +38,13 @@ pub(crate) fn store(
 
 pub(crate) fn add_protocol_set(
```
Comment (gmanev7) [inline platform/contracts/admin/src/state/contracts.rs:39]: -> `add_protocol`

**Review by gmanev7 — COMMENTED** [review]
only renaming ...

File: platform/contracts/admin/src/contracts/impl_mod.rs [hunk 4]
```diff
@@ -43,9 +43,9 @@ impl ContractsTemplate<Addr, BTreeMap<String, ProtocolWithNetworkName>> {
 
         self.protocol
             .into_iter()
-            .try_for_each(|(name, ProtocolWithNetworkName { protocol, .. })| {
+            .try_for_each(|(name, Protocol { contracts: protocol, .. })| {
```
Comment (gmanev7) [inline platform/contracts/admin/src/contracts/impl_mod.rs:46]: same as above

File: platform/contracts/admin/src/contracts/impl_mod.rs [hunk 5]
```diff
@@ -10,19 +10,19 @@ use crate::{
...
+impl ContractsTemplate<Addr, BTreeMap<String, Protocol>> {
     fn migrate(self, mut migration_msgs: ContractsMigration) -> Result<Batch> {
         let mut batch: Batch = Batch::default();
 
         self.platform.migrate(&mut batch, migration_msgs.platform);
 
         self.protocol
             .into_iter()
-            .try_for_each(|(name, ProtocolWithNetworkName { protocol, .. })| {
+            .try_for_each(|(name, Protocol { contracts: protocol, .. })| {
```
[the remark this answers is not in the document]
Comment (KirilMihaylov) [reply platform/contracts/admin/src/contracts/impl_mod.rs:25]: No, not really. The IDE must have not renamed the local use to keep it how it is there.

**Review by gmanev7 — APPROVED** [review]

**Pull request merged**